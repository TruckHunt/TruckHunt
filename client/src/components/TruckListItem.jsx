import React, { Component } from 'react';
import { Col, Label, ListGroupItem } from 'react-bootstrap';
import { Link } from 'react-router-dom';
// import { mapCenterUpdate } from '../actions/mapActions.js';


class TruckListItem extends Component {

  render() {
    const info = this.props.restaurant;
    const random = (Math.floor(Math.random() * 4) + 1);
    const image = `https://s3-us-west-1.amazonaws.com/zollstorage/MapMarker(large)V${random}.png`;

    return (
      <ListGroupItem className="trucklist-group" >
        <Col xs={12} sm={4} style={{ textAlign: 'center' }}>
          <img src={image} style={{ maxHeight: '90px' }} alt={`${info.vendor_name}-img`} />
        </Col>
        <Col xs={12} sm={8}>
          <Col xs={9} sm={8}>
            <h2>{ info.vendor_name }</h2>
            <div><h4>{ info.food_category }</h4></div>
          </Col>
          <Col xs={3} sm={3} className="trucklist-rating">
            <Label bsStyle="default" style={{ fontSize: '2.5em' }}>
              4/5
            </Label>
          </Col>
        </Col>
      </ListGroupItem>
    );
  }
}


export default TruckListItem;

