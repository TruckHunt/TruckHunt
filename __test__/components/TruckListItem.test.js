import React from 'react';
import renderer from 'react-test-renderer';
import TruckListItem from '../../client/src/components/TruckListItem.jsx';

const restaurant = {
  vendor_id: 5,
  vendor_name: 'BH & MT LLC',
  food_category: 'Cold Truck: Breakfast: Sandwiches: Salads: Pre-Packaged Snacks: Beverages',
  phone_number: null,
  day_of_week: 1,
  start_time: 11,
  end_time: 15,
  rating: 4
};

describe('CartItem Component Tests', () => {
  test('CartItem matches snapshot', () => {
    const component = renderer.create(<TruckListItem
      restaurant={restaurant}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
