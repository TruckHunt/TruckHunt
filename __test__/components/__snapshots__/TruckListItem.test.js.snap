// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truck List Irem Component Tests Truck List Item matches snapshot 1`] = `
<div
  className="ui card truckCard"
  href={undefined}
  onClick={[Function]}
>
  <img
    alt={undefined}
    className="ui fluid image"
    height={undefined}
    src="https://s3-us-west-1.amazonaws.com/zollstorage/truckhunt/square/3_3.jpg"
    width={undefined}
  />
  <div
    className="content"
  >
    <div
      className="header"
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      BH & MT LLC
    </div>
    <div
      className="meta"
    >
      Cold Truck, Breakfas
      ...
    </div>
  </div>
  <div
    className="extra content"
  >
    <div
      className="ui grid"
    >
      <div
        className="six wide column"
      >
        <div
          className="ui star rating"
          onMouseLeave={[Function]}
          role="radiogroup"
        >
          <div
            aria-checked={false}
            aria-posinset={1}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={0}
          />
          <div
            aria-checked={false}
            aria-posinset={2}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={0}
          />
          <div
            aria-checked={false}
            aria-posinset={3}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={0}
          />
          <div
            aria-checked={true}
            aria-posinset={4}
            aria-setsize={5}
            className="active icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={0}
          />
          <div
            aria-checked={false}
            aria-posinset={5}
            aria-setsize={5}
            className="icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            role="radio"
            tabIndex={0}
          />
        </div>
      </div>
      <div
        className="ten wide column"
      >
        <div
          className="ui orange label"
          onClick={[Function]}
        >
          Open 
          11 AM
           to 
          3 PM
        </div>
      </div>
    </div>
  </div>
</div>
`;
